src/include/daos_fs_sys.h

```diff
@@ -542,6 +542,19 @@ int
 dfs_sys_mkdir(dfs_sys_t *dfs_sys, const char *dir, mode_t mode,
 	      daos_oclass_id_t cid);
 
+/**
+ * Create a directory and all of its parent directories.
+ *
+ * \param[in]	dfs_sys Pointer to the mounted file system.
+ * \param[in]	dir	Link path of new dir.
+ * \param[in]	mode	mkdir mode.
+ * \param[in]	cid	DAOS object class id (pass 0 for default MAX_RW).
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_sys_mkdir_p(dfs_sys_t *dfs_sys, const char *dir, mode_t mode, daos_oclass_id_t cid);
```

Comment:
what is the reasoning for creating this at the sys level and not at the lower level dfs api level and then just add the sys wrapper?

Reply:
indeed you are correct. probably just checking dfs_mkdir EEXIST error should be sufficient and not necessary to add dfs_mkdir_p. sorry :-)

Reply:
The only public DFS function that takes a path is `dfs_lookup`, which doesn't do any creates. The main motivation for DFS Sys was that it accepts a path instead of parent object handle.
It would be a bit more work but @mchaarawi should we modify `lookup_rel_path` (internal function) to conditionally create as it walks the path?

Reply:
we can explore sure, but way out of scope for this PR.
i always forget that we have the sys api for these lookup traversals :-)

Reply:
Yes, agree it's out of scope here. We can always rework some internals later to optimize